src/tree/updater_quantile_hist.h

```diff
@@ -148,6 +149,40 @@ struct CPUHistMakerTrainParam
   }
 };
 
+/* tree growing policies */
+struct ExpandEntry {
+  static const int kRootNid  = 0;
+  static const int kEmptyNid = -1;
+  int nid;
+  int sibling_nid;
+  int depth;
+  bst_float loss_chg;
+  unsigned timestamp;
+  ExpandEntry(int nid, int sibling_nid, int depth, bst_float loss_chg,
```

Comment:
I think by now you have some familiarity with the GPU implementation, do you think it's feasible to get rid of the `sibling` in CPU implementation in the future?

Reply:
sure. I tried to avoid it. Please check it